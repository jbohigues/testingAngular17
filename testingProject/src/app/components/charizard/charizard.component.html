<h2>Pokemon</h2>

@if (!charizard) {
<h3>Loading...</h3>
} @else {
<h3>{{ charizard.name | titlecase }}</h3>
<img [src]="charizard.sprites.front_default" [alt]="charizard.name" />
}
