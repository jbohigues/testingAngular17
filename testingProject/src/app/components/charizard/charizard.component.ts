import { Component } from '@angular/core';
import { PokemonService } from '../../services/pokemon.service';
import { Pokemon } from '../../interfaces';
import { CommonModule } from '@angular/common';

@Component({
  selector: 'app-charizard',
  standalone: true,
  imports: [CommonModule],
  templateUrl: './charizard.component.html',
  styleUrl: './charizard.component.scss',
})
export class CharizardComponent {
  public charizard?: Pokemon;
  constructor(private pokemonService: PokemonService) {}

  ngOnInit() {
    this.pokemonService.getPokemon(6).subscribe((pokemon) => {
      // console.log(pokemon);
      this.charizard = pokemon;
    });
  }
}
