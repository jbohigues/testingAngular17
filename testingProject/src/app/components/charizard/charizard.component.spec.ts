import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CharizardComponent } from './charizard.component';
import { PokemonService } from '../../services/pokemon.service';
import {
  HttpClientTestingModule,
  HttpTestingController,
} from '@angular/common/http/testing';

describe('CharizardComponent', () => {
  let component: CharizardComponent;
  let fixture: ComponentFixture<CharizardComponent>;
  let compiled: HTMLElement;
  let service: PokemonService;
  let httpMock: HttpTestingController;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [CharizardComponent, HttpClientTestingModule],
      providers: [PokemonService],
    }).compileComponents();

    fixture = TestBed.createComponent(CharizardComponent);
    component = fixture.componentInstance;
    service = TestBed.inject(PokemonService);
    httpMock = TestBed.inject(HttpTestingController);

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should do match with snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });

  it('shoulkd show loading at init', () => {
    const h3 = compiled.querySelector('h3');
    expect(h3?.textContent).toContain('Loading...');
  });

  it('should charge Charizard inmediately', () => {
    const dummyPokemon = {
      name: 'charizardo!!!',
      sprites: { front_default: 'https://charizard.com/sprite.png' },
    };
    const request = httpMock.expectOne('https://pokeapi.co/api/v2/pokemon/6');
    expect(request.request.method).toBe('GET');
    request.flush(dummyPokemon);

    fixture.detectChanges();
    // console.log(compiled.innerHTML);
    const h3 = compiled.querySelector('h3');
    const img = compiled.querySelector('img');

    expect(h3?.textContent?.toLowerCase()).toContain(
      dummyPokemon.name.toLowerCase()
    );
    expect(img?.src).toBe(dummyPokemon.sprites.front_default);
    expect(img?.alt).toBe(dummyPokemon.name);
  });
});
