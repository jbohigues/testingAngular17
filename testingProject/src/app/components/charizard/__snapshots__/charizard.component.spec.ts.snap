// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharizardComponent should do match with snapshot 1`] = `"<h2>Pokemon</h2><h3>Loading...</h3><!--container--><!--container-->"`;
